bundle-workflow/src/manifests/bundle_manifest.py

```diff
@@ -45,6 +48,28 @@ def to_dict(self):
             ),
         }
 
+    @staticmethod
+    def from_s3(bucket_name, build_id, opensearch_version, architecture, work_dir=None):
+        work_dir = work_dir if not None else str(os.getcwd())
+        manifest_s3_path = BundleManifest.get_bundle_manifest_relative_location(build_id, opensearch_version, architecture)
+        S3Bucket(bucket_name).download_file(manifest_s3_path, work_dir)
+        with open('manifest.yml', 'r') as file:
+            bundle_manifest = BundleManifest.from_file(file)
+        os.remove(os.path.realpath(os.path.join(work_dir, 'manifest.yml')))
+        return bundle_manifest
+
+    @staticmethod
+    def get_tarball_relative_location(build_id, opensearch_version, architecture):
+        return f"bundles/{opensearch_version}/{build_id}/{architecture}/opensearch-{opensearch_version}-linux-{architecture}.tar.gz"
+
+    @staticmethod
+    def get_tarball_name(opensearch_version, architecture):
+        return f"opensearch-{opensearch_version}-linux-{architecture}.tar.gz"
```

Comment:
Do we need to construct the tarball name?
The bundle manifest already has the location for a given build, its something like:
```
build:
  architecture: x64
  id: '32'
  location: https://artifacts.opensearch.org/bundles/1.1.0/32/opensearch-1.1.0-linux-x64.tar.gz
  name: OpenSearch
  version: 1.1.0
```

Reply:
Yeah, either way we will have to construct it from the location or the parameters.

Reply:
Yeah instead reading from the manifest avoids having this logic baked into our code incase the format changes or new platforms are supported etc. One less problem to worry about :)

Reply:
Hey @saratvemulapalli . right now the entire codebase including build workflow is hardcoded to "linux" as platform. I think these things will gradually evolve for better, down the line, when we have more platforms getting added.